### DISCUSSION TITLE
Is it safe to copy `BL…Core` to a new address?

### DISCUSSION THREAD

**Discussion by zopsicle** [blend2d/blend2d#219] [Q&A] [answered]
Consider:

```c
BLFontDataCore font_data;
blFontDataInit(&font_data);
blFontDataCreateFromData(&font_data, ...);
BLFontDataCore font_data_2 = font_data;
UseFontData(&font_data_2);
blFontDataDestroy(&font_data_2);
```

Is such usage safe for any Blend2D object? Or do I explicitly have to call `bl…AssignMove`? I will not use the original after copying it, so this is not an issue about double frees, but rather about self-referential objects and the like.

**Comment by kobalicek** [comment, marked as answer]
Yeah, this is safe to copy and if not it would be a bug on Blend2D side.

BTW `AssignMove` API requires the destination to be already initialized (you are assigning to it), `InitMove` is an initializer (a move constructor in C++ terms).

**Reply by zopsicle** [reply]
Thanks! :+1:
